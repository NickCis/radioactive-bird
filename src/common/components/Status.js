import React from 'react';
import { Route } from 'react-router-dom';

export default ({code, children}) => (
  <Route render={({ staticContext }) => {
    if (staticContext)
      staticContext.status = code;
    return children || null;
  }} />
);
