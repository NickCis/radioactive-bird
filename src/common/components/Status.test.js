import React from 'react';
import Status from './Status';
import { StaticRouter as Router } from 'react-router';
import { mount } from 'enzyme';

describe('<Status />', () => {
  it('should set the status of the static context', () => {
    const context = {};
    const code = 404;

    mount(
      <Router context={context}>
        <Status code={code} />
      </Router>
    );

    expect(context.status).toBe(code);
  });

  it('should render childs', () => {
    const context = {};
    const code = 404;

    const wrapper = mount(
      <Router context={context}>
        <Status code={code}>
          <div>This is a test</div>
        </Status>
      </Router>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
