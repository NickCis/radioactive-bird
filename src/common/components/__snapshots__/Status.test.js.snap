// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> should render childs 1`] = `
<StaticRouter
  basename=""
  context={
    Object {
      "status": 404,
    }
  }
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component
      code={404}
    >
      <Route
        render={[Function]}
      >
        <div>
          This is a test
        </div>
      </Route>
    </Component>
  </Router>
</StaticRouter>
`;
